### DISCUSSION TITLE
htpasswd file reads and source code

### DISCUSSION THREAD

**Discussion by hardillb** [verdaccio/verdaccio#4052] [Q&A] [answered]
Hello,

I'm trying to work out when the htpasswd plugin reads the file. I  have tested by starting the Docker container with a htpasswd file mounted as a volume. I then added:

- started the container
- a new user to the htpasswd file
- logged in as the new user
- logged out
- removed the user from the htpasswd file
- I could still log in as the removed user

Is there a way to force the file to be re-read or is there a timeout that will cause it to be re-read?

Also the link to source code from the old repo 404s

https://github.com/verdaccio/verdaccio-htpasswd

Could you point me at the current httpasswd authentication plugin code.

Thanks

**Comment by juanpicado** [comment, marked as answer]
> Also the link to source code from the old repo 404s

The code is here https://github.com/verdaccio/verdaccio/tree/master/packages/plugins/htpasswd years ago was on that repo you saw, later on https://github.com/verdaccio/monorepo but those locations are not longer hosting it, check the master branch of this repo for source code.

> Is there a way to force the file to be re-read or is there a timeout that will cause it to be re-read?


I'd debug and try to find out where is the file that host the users, sometimes app might be using one and you considering is another one, here more info https://github.com/verdaccio/verdaccio/wiki/Debugging-Verdaccio

**Reply by hardillb** [reply]
Links to the mono repo are still on npmjs

**Reply by hardillb** [reply]
Follow up question, where is the source code for the AWS s3 storage plugin?

There is a directory for it in this repo, but the src/index.ts is just a place holder with a console.log() saying it's being rewritten.

Thanks.

**Reply by juanpicado** [reply]
The current code for AWS s3 plugin is here : https://github.com/verdaccio/monorepo/tree/main/plugins/aws-s3-storage but is outdated and unmaintained.

There were some initiatives to migrate to newer SDK
- https://github.com/verdaccio/verdaccio/pull/2574
- https://github.com/verdaccio/verdaccio/pull/2218

but frankly I don't have time for it, so I let it there in case someone wants to pick it up.
